airbyte-integrations/connectors/source-linkedin-ads/source_linkedin_ads/source.py

```diff
@@ -0,0 +1,357 @@
+    search_param = "search.campaign.values[0]"
+    search_param_value = "urn:li:sponsoredCampaign:"
+
+
+class AdDirectSponsoredContents(LinkedInAdsStreamSlicing):
+    """
+    Get AdDirectSponsoredContents data using `account_id` slicing.
+    More info about LinkedIn Ads / adDirectSponsoredContents:
+    https://docs.microsoft.com/en-us/linkedin/marketing/integrations/ads/advertising-targeting/create-and-manage-video?tabs=http#finders
+    """
+
+    endpoint = "adDirectSponsoredContents"
+    # AdDirectSponsoredContents stream doesn't have `id` property, so the "account" is used instead.
+    primary_key = "account"
+    parent_values_map = {"account_id": "id", "reference_id": "reference"}
+    search_param = "account"
+
+    def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = super().request_params(stream_state=stream_state, stream_slice=stream_slice, **kwargs)
+        params["owner"] = stream_slice.get("reference_id")
+        params["q"] = self.search_param
+        return params
+
+
+class LinkedInAdsAnalyticsStream(IncrementalLinkedinAdsStream):
+    """
+    AdAnalytics Streams more info:
+    https://docs.microsoft.com/en-us/linkedin/marketing/integrations/ads-reporting/ads-reporting?tabs=curl#ad-analytics
+    """
+
+    endpoint = "adAnalyticsV2"
+    # For Analytics streams the primary_key is the entity of the pivot [Campaign URN, Creative URN, etc] + `end_date`
+    primary_key = ["pivotValue", "end_date"]
+    cursor_field = "end_date"
+
+    @property
+    def base_analytics_params(self) -> MutableMapping[str, Any]:
+        """ Define the base parameters for analytics streams """
+        return {"q": "analytics", "pivot": self.pivot_by, "timeGranularity": "DAILY"}
+
+    def request_params(self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, **kwargs) -> MutableMapping[str, Any]:
+        params = self.base_analytics_params
+        params[self.search_param] = f"{self.search_param_value}{stream_slice.get(self.primary_slice_key)}"
+        params.update(**update_analytics_params(stream_slice))
+        return params
+
+    def read_records(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Optional[Mapping[str, Any]] = None, **kwargs
+    ) -> Iterable[Mapping[str, Any]]:
+        stream_state = stream_state or {self.cursor_field: self.start_date}
+        parent_stream = self.parent_stream(config=self.config)
+        result_chunks = []
+        for record in parent_stream.read_records(**kwargs):
+            for analytics_slice in make_analytics_slices(record, self.parent_values_map, stream_state.get(self.cursor_field)):
+                child_stream_slice = super().read_records(stream_slice=analytics_slice, **kwargs)
+                result_chunks.append(child_stream_slice)
+            yield from merge_chunks(result_chunks, self.cursor_field)
```

Comment:
it looks incorrect, we should reset 'result_chunks' for each outer loop or move 'yield' from the loop

Reply:
We start with the empty `result_chunks` for each new cycle. There is no need to reset anything in the middle. Or I didn't got your point here.

Reply:
Ok, we will reset the `result_chunks` for each new cycle. Done.